sys/riscv/riscv/pmap.c

```diff
@@ -2423,15 +2432,25 @@ pmap_fault_fixup(pmap_t pmap, vm_offset_t va, vm_prot_t ftype)
 	}
 
 	if ((pmap != kernel_pmap && (oldpte & PTE_U) == 0) ||
-	    (ftype == VM_PROT_WRITE && (oldpte & PTE_W) == 0) ||
+	    (((ftype & VM_PROT_WRITE) != 0) && (oldpte & PTE_W) == 0) ||
 	    (ftype == VM_PROT_EXECUTE && (oldpte & PTE_X) == 0) ||
 	    (ftype == VM_PROT_READ && (oldpte & PTE_R) == 0))
 		goto done;
 
+#if __has_feature(capabilities)
+	if (((ftype & VM_PROT_WRITE_CAP) != 0) && ((oldpte & PTE_CW) == 0))
+		goto done;
+#endif
```

Comment:
```suggestion
#if __has_feature(capabilities)
	    ((ftype & VM_PROT_WRITE_CAP) != 0 && (oldpte & PTE_CW) == 0) ||
#endif
	    (ftype == VM_PROT_EXECUTE && (oldpte & PTE_X) == 0) ||
	    (ftype == VM_PROT_READ && (oldpte & PTE_R) == 0))
		goto done;
```
is shorter and clearer